lib/dry/auto_inject/injector.rb

```diff
@@ -2,22 +2,34 @@
 
 module Dry
   module AutoInject
-    class Injector
+    class Injector < BasicObject
       # @api private
       attr_reader :container
 
       # @api private
       attr_reader :strategy
 
       # @api private
-      def initialize(container, strategy)
+      attr_reader :builder
+
+      # @api private
+      def initialize(container, strategy, builder:)
         @container = container
         @strategy = strategy
+        @builder = builder
       end
 
       def [](*dependency_names)
         strategy.new(container, *dependency_names)
       end
+
+      def method_missing(name, *args, &block)
+        builder.public_send(name)
+      end
+
+      def respond_to?(name, include_private = false)
+        name == :[] || builder.respond_to?(name, include_private)
```

Comment:
`respond_to?` is actually not related to `method_missing` so it seems like a good idea to pass in include_private here, I think O_o